### PR TITLE
update: mysql connector/j update for debugging

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.6 (TiDB 7.6 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by jackysp — APPROVED** [review]

**Comment by dveeden** [comment]
It looks like MySQL Connector/J 8.0 doesn't get any updates anymore. The latest G.A. of MySQL Connector/J is 8.2 (while MySQL Server 8.2 isn't GA, which is confusing).

I think we should recommend or at least mention MySQL Connector/J 8.2.

**Review by dveeden — APPROVED** [review]

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-7.3 label because the v7.3 docs have been archived at https://docs-archive.pingcap.com/tidb/v7.3 and will no longer receive new updates.

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 1]
```diff
@@ -153,27 +153,27 @@ To ensure data consistency by transaction, you can use `UPDATE` statements to up
...
+> `useConfigs=maxPerformance` includes a group of configurations. To get the detailed configurations in MySQL Connector/J 8.0 and those in MySQL Connector/J 5.1, see [mysql-connector-j 8.0](https://github.com/mysql/mysql-connector-j/blob/release/8.0/src/main/resources/com/mysql/cj/configurations/maxPerformance.properties) and [mysql-connector-j 5.1](https://github.com/mysql/mysql-connector-j/blob/release/5.1/src/com/mysql/jdbc/configs/maxPerformance.properties) respectively. Please disable `useLocalTransactionState` when using `maxPerformance`. That is, use `useConfigs=maxPerformance&useLocalTransactionState=false`.
 
-**DO NOT** turn on `useLocalTransactionState` as this might prevent transactions from being committed or rolled back.
+This bug has been fixed in MySQL Connector/J version `8.0.33`. Please use version `8.0.33` or a higher version. For improved stability and performance, and considering that updates for the `8.0` series have ceased, we strongly recommend upgrading to the latest General Availability (GA) version, which is currently `8.2`.
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:164]:
```suggestion
This bug has been fixed in MySQL Connector/J version `8.0.33`. Please use version `8.0.33` or a higher version. For improved stability and performance, and considering that updates for the `8.0` series have ceased, we strongly recommend upgrading to the latest General Availability (GA) version, which is currently `8.3`.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 2]
```diff
@@ -153,27 +153,27 @@ To ensure data consistency by transaction, you can use `UPDATE` statements to up
...
 
 **Description**
 
-The database connection might hang under certain conditions when using MySQL Connector/J 8.0.29 with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
+The database connection might hang under certain conditions when using MySQL Connector/J `8.0.31` and below with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
 
 **Way to avoid**
 
-This is a known issue. As of October 12, 2022, MySQL Connector/J has not fixed the issue.
+This bug has been fixed in MySQL Connector/J version `8.0.32`. Please use version `8.0.32` or a higher version. For improved stability and performance, and considering that updates for the `8.0` series have ceased, we strongly recommend upgrading to the latest General Availability (GA) version, which is currently `8.2`.
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:174]:
```suggestion
This bug has been fixed in MySQL Connector/J version `8.0.32`. Please use version `8.0.32` or a higher version. For improved stability and performance, and considering that updates for the `8.0` series have ceased, we strongly recommend upgrading to the latest General Availability (GA) version, which is currently `8.3`.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 3]
```diff
@@ -153,27 +153,27 @@ To ensure data consistency by transaction, you can use `UPDATE` statements to up
...
 
-**DO NOT** turn on `useLocalTransactionState` as this might prevent transactions from being committed or rolled back.
+This bug has been fixed in MySQL Connector/J 8.0.33. Considering updates for the 8.0.x series have ceased, it is strongly recommended to upgrade your MySQL Connector/J to the latest General Availability (GA) version (currently 8.3) for improved stability and performance.
 
 ### Connector is incompatible with the server version earlier than 5.7.5
 
 **Description**
 
-The database connection might hang under certain conditions when using MySQL Connector/J 8.0.29 with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
+The database connection might hang under certain conditions when using MySQL Connector/J 8.0.31 an earlier version with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
```
Comment (qiancai) [inline develop/dev-guide-third-party-tools-compatibility.md:170]:
```suggestion
The database connection might hang under certain conditions when using MySQL Connector/J 8.0.31 or an earlier version with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
```

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-7.4 label because the v7.4 docs have been archived at https://docs-archive.pingcap.com/tidb/v7.4 and will no longer receive new updates.

**Comment by Icemap** [comment]
Should we merge it? @qiancai

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 4]
```diff
@@ -153,27 +153,27 @@ To ensure data consistency by transaction, you can use `UPDATE` statements to up
...
 
 **Way to avoid**
 
 > **Note:**
 >
-> This bug has been reported to MySQL JDBC. To keep track of the process, you can follow this [Bug Report](https://bugs.mysql.com/bug.php?id=108643).
+> `useConfigs=maxPerformance` includes a group of configurations. For detailed configurations in MySQL Connector/J 8.0 and MySQL Connector/J 5.1, see [mysql-connector-j 8.0](https://github.com/mysql/mysql-connector-j/blob/release/8.0/src/main/resources/com/mysql/cj/configurations/maxPerformance.properties) and [mysql-connector-j 5.1](https://github.com/mysql/mysql-connector-j/blob/release/5.1/src/com/mysql/jdbc/configs/maxPerformance.properties) respectively. You need to disable `useLocalTransactionState` when using `maxPerformance`. That is, use `useConfigs=maxPerformance&useLocalTransactionState=false`.
 
-**DO NOT** turn on `useLocalTransactionState` as this might prevent transactions from being committed or rolled back.
+This bug has been fixed in MySQL Connector/J 8.0.33. Considering updates for the 8.0.x series have ceased, it is strongly recommended to upgrade your MySQL Connector/J to the latest General Availability (GA) version (currently 8.3) for improved stability and performance.
```
Comment (qiancai) [inline develop/dev-guide-third-party-tools-compatibility.md:164]:
```suggestion
This bug has been fixed in MySQL Connector/J 8.0.33. Considering updates for the 8.0.x series have ceased, it is strongly recommended to upgrade your MySQL Connector/J to [the latest General Availability (GA) version](https://dev.mysql.com/downloads/connector/j/) for improved stability and performance.
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 5]
```diff
@@ -153,27 +153,27 @@ To ensure data consistency by transaction, you can use `UPDATE` statements to up
...
 
 **Description**
 
-The database connection might hang under certain conditions when using MySQL Connector/J 8.0.29 with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
+The database connection might hang under certain conditions when using MySQL Connector/J 8.0.31 or an earlier version with a MySQL server < 5.7.5 or a database using the MySQL server < 5.7.5 protocol (such as TiDB earlier than v6.3.0). For more details, see the [Bug Report](https://bugs.mysql.com/bug.php?id=106252).
 
 **Way to avoid**
 
-This is a known issue. As of October 12, 2022, MySQL Connector/J has not fixed the issue.
+This bug has been fixed in MySQL Connector/J 8.0.32. Considering updates for the 8.0.x series have ceased, it is strongly recommended to upgrade your MySQL Connector/J to the latest General Availability (GA) version (currently 8.3) for improved stability and performance.
```
Comment (qiancai) [inline develop/dev-guide-third-party-tools-compatibility.md:174]:
```suggestion
This bug has been fixed in MySQL Connector/J 8.0.32. Considering updates for the 8.0.x series have ceased, it is strongly recommended to upgrade your MySQL Connector/J to [the latest General Availability (GA) version](https://dev.mysql.com/downloads/connector/j/) for improved stability and performance.
```

File: develop/dev-guide-third-party-tools-compatibility.md (hunk 4 above)
Comment (qiancai) [reply develop/dev-guide-third-party-tools-compatibility.md:164]: reason: avoid updating the version again when there are newer GA versions

**Comment by qiancai** [comment]
@Icemap Rest LGTM.

**Pull request merged**